deployment/deployment.go

```diff
@@ -0,0 +1,85 @@
+package deployment
+
+import (
+	"os"
+
+	"github.com/akitasoftware/akita-cli/printer"
+	"github.com/akitasoftware/akita-libs/tags"
+)
+
+type Deployment string
+
+const (
+	None       Deployment = ""
+	Unknown               = "unknown"
+	Kubernetes            = "kubernetes"
+)
+
+func (d Deployment) String() string {
+	return string(d)
+}
+
+func GetDeploymentInfo() (Deployment, map[tags.Key]string) {
+	// Allow the user to specify a deployment environment, even
+	// if it's of an unknown type.
+
+	deploymentType := None
+	tagset := make(map[tags.Key]string)
+
+	if d := os.Getenv("AKITA_DEPLOYMENT"); d != "" {
+		deploymentType = Unknown
+		tagset[tags.XAkitaDeployment] = d
+	}
+
+	// If there is a git commit associated with this deployment,
+	// then record it.
+	if c := os.Getenv("AKITA_DEPLOYMENT_COMMIT"); c != "" {
+		tagset[tags.XAkitaGitCommit] = c
+	}
+
+	if GetAWSTags(tagset) {
+		printer.Infof("Found AWS environment variables.\n")
+	}
+
+	if GetKubernetesTags(tagset) {
+		printer.Infof("Found Kubernetes environment variables.\n")
+		deploymentType = Kubernetes
```

Comment:
This overwrites the deployment type obtained from `AKITA_DEPLOYMENT` in the environment. Is this intentional?

Reply:
I wish I had come up with a synonym. Any suggestions?

The user can't set the deploymentType, which is only used internally; it doesn't even appear as a tag (though maybe the lesson is that it should!) I figured the presence of Kubernetes-specific tags was sufficient.

They can set the x-akita-deployment value, which will be "staging" or "production" or "test1" etc. to distinguish different clusters.

Reply:
Looks like this is no longer an issue with this rewrite.